server/sms/sms_gateway.go

```diff
@@ -189,10 +189,12 @@ func (g *gateway) send(receivedMsg *protocol.Message) error {
 	if err != nil {
 		log.WithField("error", err.Error()).Error("Sending of message failed")
 		mTotalResponseErrors.Add(1)
+		pNexmoResponseErrors.Inc()
 		return err
 	}
-	mTotalSentMessages.Add(1)
 	g.SetLastSentID(receivedMsg.ID)
+	mTotalSentMessages.Add(1)
```

Comment:
Shouldn't this be before the `SetLastSentID` because the message was sent actually.